models/issues/issue.go

```diff
@@ -2498,3 +2498,20 @@ func DeleteOrphanedIssues(ctx context.Context) error {
 func (issue *Issue) HasOriginalAuthor() bool {
 	return issue.OriginalAuthor != "" && issue.OriginalAuthorID != 0
 }
+
+type IssueCloseState int
+
+const (
+	// IssueStateOpen
+	IssueStateOpen IssueCloseState = iota
+	// IssueStateLocked
+	IssueStateLocked
```

Comment:
I am not sure whether this is the final design (I see this PR is WIP), but I'd like to have quick comment about this design: I guess the "Locked" should be a separate state. For example: Merged, Merged+Locked